internal/controller/clusterinstance/template_engine.go

```diff
@@ -42,6 +43,92 @@ type TemplateEngine struct {
 	Log logr.Logger
 }
 
+type Action string
+
+const (
+	Render   Action = "render"
+	Suppress Action = "suppress"
+	Prune    Action = "prune"
+)
+
+type RenderedObject struct {
```

Comment:
Might be cleaner to give this its own file.

Reply:
Good suggestion! Done